### PR TITLE
[CELEBORN-1525] Bump Ratis version from 3.1.0 to 3.1.1

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Bump Ratis version from 3.1.0 to 3.1.1 including:

- Remove `address2String` and use `setAddress(ratisAddr)` with the release of https://github.com/apache/ratis/pull/1125.
- Support `raft.grpc.message.size.max` must be 1m larger than `raft.server.log.appender.buffer.byte-limit` for https://github.com/apache/ratis/pull/1132.

### Why are the changes needed?

Bump Ratis version from 3.1.0 to 3.1.1. Ratis has released v3.1.1, of which release note refers to [3.1.1](https://ratis.apache.org/post/3.1.1.html). The 3.1.1 version is a minor release with multiple improvements and bugfixes including [[RATIS-2116] Fix the issue where RaftServerImpl.appendEntries may be blocked indefinitely](https://issues.apache.org/jira/browse/RATIS-2116), [[RATIS-2131] Configuring Ratis fails when hostname is used, and is an IPv6 host](https://issues.apache.org/jira/browse/RATIS-2131). See the [changes between 3.1.0 and 3.1.1](https://github.com/apache/ratis/compare/ratis-3.1.0...ratis-3.1.1) releases.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI.

### CODE REVIEW

**Comment by mridulm** [comment]
In description, diff of changes between the two versions, would be https://github.com/apache/ratis/compare/release-3.1.0...release-3.1.1 ?
Our changes look good to me.

**Comment by SteNicholas** [comment]
@mridulm, Ratis 3.1.1 version will release this week. Therefore, I converted this pull request to draft. Meanwhile, I used Ratis v3.1.1-rc3 version to test this bumping firstly.

**Comment by mridulm** [comment]
The test failures look related to RATIS-2135

**Comment by SteNicholas** [comment]
> The test failures look related to RATIS-2135

@mridulm, I have supported that `raft.grpc.message.size.max` must be 1m larger than `raft.server.log.appender.buffer.byte-limit` for RATIS-2135. PTAL.

File: pom.xml [hunk 1]
```diff
@@ -94,7 +94,7 @@
     <netty.version>4.1.109.Final</netty.version>
     <bouncycastle.version>1.77</bouncycastle.version>
     <protobuf.version>3.21.7</protobuf.version>
-    <ratis.version>3.1.0</ratis.version>
+    <ratis.version>3.1.1</ratis.version>
```
Comment (cxzl25) [inline pom.xml:97]: 3.1.1 fixed a very serious bug. In the cluster I use, the Master and Follower status often appear to be out of sync, so upgrading to 3.1.1 looks good.

RATIS-2116. Fix the issue where RaftServerImpl.appendEntries may be blocked indefinitely

https://issues.apache.org/jira/browse/RATIS-2116

Comment (SteNicholas) [reply pom.xml:97]: @cxzl25, I have mentioned this in the description of pull request.

File: master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HARaftServer.java [hunk 2]
```diff
@@ -200,9 +200,7 @@ public static HARaftServer newMasterRatisServer(
     RaftPeer localRaftPeer =
         RaftPeer.newBuilder()
             .setId(localRaftPeerId)
-            // TODO(CELEBORN-1525): Once Ratis 3.2.0 is released and we move to it, we
-            // should remove address2String and change this to .setAddress(ratisAddr)
-            .setAddress(address2String(ratisAddr))
+            .setAddress(ratisAddr)
```
Comment (cxzl25) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HARaftServer.java:203]: Should the `address2String` method also be removed?

**Review by cxzl25 — APPROVED** [review]

**Review by mridulm — APPROVED** [review]
LGTM !

**Comment by mridulm** [comment]
Merged to master.
Thanks for fixing it @SteNicholas !
Thanks for reviews @cxzl25 , @szetszwo :-)

**Pull request closed without merging**